android/src/main/kotlin/com/apptentive/apptentive_flutter/ApptentiveFlutterPlugin.kt

```diff
@@ -325,7 +342,7 @@ class ApptentiveFlutterPlugin : FlutterPlugin, MethodCallHandler, ActivityAware
             channel.invokeMethod("onSurveyFinished", mapOf("completed" to true))
           }
 
-        interaction == "Survey" && name != "submit" ->
+        interaction == "Survey" && name == "cancel" || name == "cancel_partial" ->
```

Comment:
@ChaseApptentive thought about this while writing the documentation. The check `!= "submit"` would trigger the `onSunrveyFinished` with `completed` as `false` on the survey launch. That wouldn't make sense as it is a launch event. So added exact check of "cancel" and "cancel_partial".

Reply:
I still think this should only be returning when a survey is submitted. So it'll only ever return true. Otherwise they should assume it's false.

Reply:
@frankus @ChaseApptentive legacy android did this way. Also, iOS currently supporting the survey listener the same way. 

    case "com.apptentive#Survey#submit":
      self.channel.invokeMethod("onSurveyFinished", arguments: ["completed": true])
    case "com.apptentive#Survey#cancel", "com.apptentive#Survey#cancel_partial":
      self.channel.invokeMethod("onSurveyFinished", arguments: ["completed": false])

If the App has been using or expecting a survey listener work this way, we shouldn't be probably changing it.